### PR TITLE
Home Sections Feature + Review Mode

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#12532

*Description of changes:*
* Part Two of the Home Sections
* Includes full Home Section feature 
* Add / Edit form
* Table for Display with Action features
* Review mode of Table
* Basic Structure of Review Confirm view for MHR Registrations

* Part 3(To come in a follow up commit) is UNIT TESTING, in order to get this to UX soon as possible.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue [hunk 1]
```diff
@@ -1,83 +1,129 @@
...
       />
     </v-expand-transition>
 
     <!-- Home Sections Table -->
     <article class="mt-6">
-      <p>Number of Sections: 0</p>
-      <HomeSectionsTable />
+      <v-row no-gutters>
+        <p v-if="!isReviewMode">Number of Sections: {{getMhrHomeSections.length}}</p>
+        <span v-if="false && hasMinimumHomeSections" class="pl-4 error-text">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue:33]: `false` placeholder for App wide validations when included

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue [hunk 2]
```diff
@@ -1,83 +1,129 @@
...
+          Your registration must contain at least one section
+        </span>
+      </v-row>
+      <HomeSectionsTable
+        :class="{ 'border-error-left': false }"
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue:38]: `false` flag placeholder for app level validations when this implemented.

File: ppr-ui/src/components/registration/RegistrationBarTypeAheadList.vue [hunk 3]
```diff
@@ -98,7 +98,7 @@ export default defineComponent({
       isDense: props.defaultDense,
       isClearable: props.defaultClearable,
       includeMhrSelection: computed((): boolean => {
-        return isRoleStaff.value && getFeatureFlag('mhr-ui-enabled')
+        return isRoleStaff.value && getFeatureFlag('mhr-registration-enabled')
```
Comment (cameron-eyds) [inline ppr-ui/src/components/registration/RegistrationBarTypeAheadList.vue:101]: New feature flag for MHR Registrations, for better release control

File: ppr-ui/src/views/newMhrRegistration/MhrReviewConfirm.vue [hunk 4]
```diff
@@ -1,18 +1,133 @@
...
+import { HomeSections } from '@/components/mhrRegistration/YourHome'
+import { RouteNames } from '@/enums'
+/* eslint-enable no-unused-vars */
 
 @Component({
   components: {
+    HomeSections
   }
 })
 export default class MhrReviewConfirm extends Vue {
```
Comment (cameron-eyds) [inline ppr-ui/src/views/newMhrRegistration/MhrReviewConfirm.vue:112]: A lot of placeholders in this view, just trying to set up the basic framework for Review version of components.

**Review by doug-lovett — APPROVED** [review]
LGTM.

File: ppr-ui/src/views/newMhrRegistration/MhrRegistration.vue [hunk 5]
```diff
@@ -132,21 +128,19 @@ export default class MhrRegistration extends Vue {
     // do not proceed if app is not ready
     if (!val) return
     // redirect if not authenticated (safety check - should never happen) or if app is not open to user (ff)
-    if (!this.isAuthenticated || (!this.isJestRunning && !getFeatureFlag('mhr-ui-enabled'))) {
+    if (!this.isAuthenticated || (!this.isJestRunning && !getFeatureFlag('mhr-registration-enabled'))) {
```
Comment (dimak1) [inline ppr-ui/src/views/newMhrRegistration/MhrRegistration.vue:131]: Just to clarify, the requirement is to have the ability to Register but not Search for MHR? So `mhr-ui-enabled` is not a global flag for MHR? We can still see different functionalities (i.e. registration) even if we have disabled MHR UI ?

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue [hunk 6]
```diff
@@ -1,83 +1,129 @@
...
+      'getMhrHomeSections'
     ])
     const localState = reactive({
+      isEditingHomeSection: false,
       isNewHomeSection: true,
-      showAddEditHomeSections: false
+      displayHomeSectionsError: false,
+      showAddEditHomeSections: false,
+      isMaxHomeSections: computed((): boolean => {
+        return getMhrHomeSections.value.length === 4
```
Comment (dimak1) [inline ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue:89]: Should this be less-than-equal 4?

**Review by dimak1 — APPROVED** [review]
Looks good, just a few comments.

**Comment by dimak1** [comment]
Nice work! 🚀

File: ppr-ui/src/views/newMhrRegistration/MhrRegistration.vue (hunk 5 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/newMhrRegistration/MhrRegistration.vue:131]: Generally, the requirement will be to enable the MHR Search prior to enabling Registrations, which is something we would have bumped into when we the time came to enable `mhr-ui-enabled` for Staff to do searches. 
This just allows us to release those features to staff while protecting our in-progress Registrations flow. 

Your statement is correct though, `mhr-ui-enabled` will not be global for all mhr features.

We could have repurposed the original flag at the point of release, but would be safer to release with BOTH flags, in the event we wanted to 'shut-off' the search features for whatever reason.

File: ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue (hunk 6 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/YourHome/HomeSections.vue:89]: In this context, we want to identify when the user has reached their max home sections, which is 4. 
Once they have added 4 home sections. they are disabled from adding any further, as well as an warning messaging being prompt, if they are at 4 sections. 

This number is something we can update to a dynamic value at a later date, if we had additional rules or restrictions removed where a user could have more.

**Pull request merged**